tinymce-single-react-ui/src/Turducken.js

```diff
@@ -0,0 +1,82 @@
+import React, { createElement, Component } from 'react'
+import { render } from 'react-dom'
+import * as Icons from './external/dashicons/index'
+import BlockChangeToolbar from './components/toolbar/BlockChangeToolbar'
+import BlockToolbar from './components/toolbar/BlockToolbar'
+import InlineToolbar from './components/toolbar/InlineToolbar'
+import TinyMCEReact from './components/tinymce/tinymce-react-ui'
+import {blockList, blockType, blockAlign, getTopLevelBlock} from './utils/tag'
+import '../../shared/post-content'
+import 'assets/stylesheets/main.scss'
+
+let blockOpen  = (collapsed) => (collapsed !== null &&  collapsed) // block  if caret
+let inlineOpen = (collapsed) => (collapsed !== null && !collapsed) // inline if range selection
+
+// get tiny node from the container, and the top level block from the caret node
+let tinyNode = (containerNode) => ((containerNode && containerNode.children.length > 0) ? containerNode.children[0] : null)
+let topLevelBlock = (tinyNode, node) => ((tinyNode && node) ? getTopLevelBlock(tinyNode, node) : null)
+
+// Rect at the start of the Range
+let findStartOfRange = (range) => {
+	// make a collapsed range at the start point
+	if (range) {
+		let r = range.cloneRange();
+		r.setEnd(range.startContainer, range.startOffset);
+		return r.getBoundingClientRect();
+	}
+}
+
+// Rect for the Range
+let rangeRect = (range) => {
+	if (range) {
+		return range.getBoundingClientRect();
+	}
+}
+
+export default function Turducken(props) {
+  let store = props.store
+  let collapsed = store.getState().get('collapsed')
+  let range = store.getState().get('range')
+  let node =  store.getState().get('node') // node of caret or ancestor of range
+  let editorRef = store.getState().get('editorRef')
+  let tiny = tinyNode(editorRef)
+  let topBlock = topLevelBlock(tiny, node)
+  let topRect = rangeRect(topBlock)
+
+  return (
+    <div>
+      { topRect && topRect.width &&
+        <div>
+          <div style={ {position: 'absolute', top: topRect.top + 'px',
+                                              width: topRect.width + 'px',
+                                              height: '2px', background: '#ccc' } }></div>
+
+          <div style={ {position: 'absolute', top: topRect.height + topRect.top + 'px', width: topRect.width + 'px', height: '2px', background: '#ccc' } }></div>
+
+          <div style={ {position: 'absolute', top: topRect.top + 'px', width: '2px', height: topRect.height + 'px', background: '#ccc' } }></div>
+
+          <div style={ {position: 'absolute', top: topRect.top + 'px', width: '2px', height: topRect.height + 'px', left: topRect.width + topRect.left + 'px', background: '#ccc' } }></div>
+
+        </div>
+      }
+      <div>
+        <InlineToolbar isOpen={ inlineOpen(collapsed) }
```

Comment:
@youknowriad  we split these two things to explore an idea of @annaephox.
So that  inline editing options are be presented to the user directly at any range selection in the same way as medium, while leaving the block-level operations attached to the block.

We (@mimo84 and I) only just started on this project and didnt understand the blocks API requirement until this PR :)